asubFunctionsApp/src/copyArgA.c

```diff
@@ -30,12 +30,12 @@ static long copyArgA(aSubRecord *prec)
          errlogPrintf("%s incorrect input type. ", prec->name);
 		 return -1;
 	}
+    
 	memset(str_out, '\0', prec->nova); /* pad with NULL */
-    for(n=0; n<prec->noa && n<prec->nova; ++n)
+    for(n=0; n<prec->noa && n<prec->nea; ++n)
     {
         str_out[n] = str_in[n];
     }
-    prec->neva = (prec->nea < n ? prec->nea : n);
```

Comment:
I am not sure now, I think it was part of the reason the unwanted characters appeared, but you are right. I have added it back in with some modifications: neva now is set to n(length of input string)+1(needed for null terminator) with nova as maximum length.